tartiflette/engine.py

```diff
@@ -322,33 +390,17 @@ def __init__(
         :return: computed response corresponding to the request
         :rtype: AsyncIterable[Dict[str, Any]]
         """
-        # pylint: disable=too-many-locals
+
         document, errors = parse_and_validate_query(query, self._schema)
-        if errors:
-            yield await self._build_response(errors=errors)
-            return
 
-        source_event_stream = await create_source_event_stream(
+        # Goes through potential schema directives and finish in self._perfom_subscription
+        async for payload in self._subcription_executor(
             self._schema,
             document,
-            self._build_response,
-            initial_value,
+            errors,
+            operation_name,
             context,
             variables,
-            operation_name,
-        )
-
-        if isinstance(source_event_stream, dict):
-            yield source_event_stream
-            return
-
-        async for payload in source_event_stream:
-            yield await execute(
-                self._schema,
-                document,
-                self._build_response,
-                payload,
-                context,
-                variables,
-                operation_name,
-            )
+            initial_value,
```

Comment:
You should filled in the `context_coercer` parameter in order to forward it to potentials arguments directives:

```suggestion
            initial_value,
            context_coercer=context,
```